### DISCUSSION TITLE
Dreamhost Installation

### DISCUSSION THREAD

**Discussion by aoirthoir** [joelesko/tht#18] [General]
Dreamhost defaults to an earlier version on which resides at

/usr/bin/php

To fix, I created a bin directory in my home. You could create it anywhere if you want in your home... just make sure it is on your path before /usr/bin. Then you should be able to install and run tht. 

so these steps

```console
user@dreamhost:~$ mkdir bin
user@dreamhost:~$ cd bin
user@dreamhost:~/bin$ ln -s /usr/local/php82/bin/php
user@dreamhost:~/bin$ PATH=~/bin:$PATH
user@dreamhost:~/bin$ cd
user@dreamhost:~/$ tht

-------------
     THT     
-------------

- Version: 0.7.1 - Beta

- Commands:

  · new <appName>   create an app in the current dir
  · fix             clear app cache and update file permissions
  · info            get detailed config and install info
  · server          start the local test server (port: 3333)
  · server <port>   start the local test server on a custom port

> Usage: tht [command]

user@dreamhost:~/$
```

set your path to ~/bin/:$PATH

then THT will run 

(there are other errors which I will update in here when I validate if they are dreamhost specific or general errors. Probably on the weekend.)

**Comment by joelesko** [comment]
That's interesting.  I actually run THT on dreamhost and don't remember doing anything extra. I'll review this when I get a chance.

**Comment by aoirthoir** [comment]
For some reason even tho in my dreamhost panel i have it set to 8.x version of php, in the cli it only gives me 7.x.

So I installed tht and that worked...

in /home/username/bin is the softlink to php8

tht is instlalled in /home/username/tht 

and when i run tht new appname, from any directory /home/username/whatever...I get this message:

-----------------
     NEW APP     
-----------------

- App name: testing

> Create new THT app in current directory? (y/N)? y
<!> Sorry, something went wrong.

  Parent shortcut `..` not allowed in path: `/home/username/tht/lib/core/modes/../../../sites/starter/config/app.jcon`

> Move or delete your app directories before trying again:

  /home/username/tasting/blabla/more/testing

The appname directory is created...  (in the case above, tht new testing...) and the config directory but nothing else is created. 

Just documenting it here, I will look at it later.

**Comment by aoirthoir** [comment]
So I could not get the tht command line to work. However i realized that i could just copy and that is what I did and was able to get the starter site ...so this is what I did...

I had installed the tht via curl but now cannot get it to do so on my other accounts. no issue I have the zip file so unzipped than wran these commands in shell:

```fish
cd ~
mv tht/sites ./
rmdir sites/system/tht
cp -av sites/starter ./THTmyDomain
cp -av tht THTmyDomain/system
mv myDomain.com myDomain.old
ln -s THTmyDomain/public myDomain.com
```

then in my browser go to myDomain.com and it works. 

To be honest i dont really mind whether the command line part works or not as long as the rest does. So now tomorrow I will start testing with an actual project.
